### PR TITLE
Adding Travis support

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

**Comment by certik** [comment]
@junghans, this looks good to me, I think it can be merged.

**Comment by cmsquared** [comment]
This looks fine.  We should eventually put gh-pages stuff back in so that our docs are auto-deployed.

**Comment by cmsquared** [comment]
@junghans are you done or do you want me to wait on merge?

**Comment by junghans** [comment]
@cmsquared: done for now.

File: app/simple_mesh_app/CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,11 @@ POSSIBILITY OF SUCH DAMAGE.
 add_executable(simple_mesh_app simple_mesh_app.cc)
 target_link_libraries(simple_mesh_app portage ${EXTRA_LIBS})
 
+if (ENABLE_MPI)
+  target_link_libraries(simple_mesh_app ${MPI_CXX_LIBRARIES})
+  set_target_properties(simple_mesh_app PROPERTIES
+                        LINK_FLAGS "${MPI_CXX_LINK_FLAGS}")
+endif (ENABLE_MPI)
```
Comment (junghans) [inline app/simple_mesh_app/CMakeLists.txt:56]: @angelaherring / @cmsquared : with this change setting the compiler to `mpicc` at cmake level is not really necessary anymore, even though it doesn't hurt.

So the ReadMe needs an update.

In the next cmake clean up we should add:
```
target_link_libraries(portage ${MPI_CXX_LIBRARIES})
```
to link `libportage` against mpi and get rid of all the above blocks.

**Pull request merged**